modules/cog-fdo-shell.c

```diff
@@ -92,8 +95,31 @@ cog_fdo_shell_class_finalize (CogFdoShellClass *klass)
 static void
 cog_fdo_shell_init (CogFdoShell *shell)
 {
+    TRACE ("");
+}
+
+gboolean
+cog_fdo_shell_initable_init (GInitable *initable,
```

Comment:
The code executed during the initialization of the cog_fdo_shell.

This [code](https://github.com/Igalia/cog/pull/163/files?file-filters%5B%5D=.c&file-filters%5B%5D=.txt&hide-deleted-files=true#diff-b0e4f3f94ad14a1e344cae725bb1cd54R687) has to be relocated here.

Reply:
I saw in some `gio` classes that implement GInitable and have different entry points, that it is also possible to just keep an internal error reference from those entry points if something fails and just use the `initable_init` method to check if something went wrong and then return `FALSE` in that case. I think in our case, that we have only one entry point, it makes more sense to just move the code here as you say.